accepted/2024/NuGet-Audit-for-PackageDownload.md

```diff
@@ -0,0 +1,106 @@
+# ***Enhancing NuGet Audit for Secure and Selective `<PackageDownload>` Vulnerability Reporting***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- Author Name [Nigusu](https://github.com/Nigusu-Allehu)
+- [GitHub Issue](https://github.com/NuGet/Home/issues/13658)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+In our effort to strengthen the security of NuGet, we would like to expand the support of NuGet Audit to `<PackageDownload>` packages. This will be done by introducing vulnerability errors for packages downloaded using `PackageDownload`. However, the .Net sdk also makes use of `PackageDownload` to implicitly download packages. And we would like to make sure these packages are not also receiving these errors, in order to prevent customers being confused. This will be done by introducing a new attribute to `PackageDownload` that makes it clear to NuGet Audit it should not write an error for these packages.
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+We are expanding NuGet Audit to `<PackageDownload>` packages to improve security notifications for user-specified dependencies, addressing gaps in vulnerability alerts that currently miss these packages. This change prevents confusion by excluding SDK-managed PackageDownloads from error reports, ensuring users only see relevant security issues. The expected outcome is a more precise and useful vulnerability reporting experience, enabling users to address actual risks without unnecessary SDK-related alerts.
```

Comment:
> This change prevents confusion by excluding SDK-managed PackageDownloads from error reports, ensuring users only see relevant security issues.

The SDK adding a PackageDownload for a package with a known vulnerability is relevant. This is the purpose of Audit, to inform the customer that packages with known vulnerabilities are being used.

Unless customers get notified some other way that they're using an out of date SDK, that is including content with known vulnerabilities in their builds, I disagree fundamentally with this proposal.

Reply:
Regarding PackageDownload from the SDK here is the idea
> We should come up with an experience that notifies the user that the SDK is out of date and that they may be vulnerable to security issues. This experience should probably not even refer to PackageDownloads, as that’s an implementation detail of the SDK. We could have the same or similar experience for notifying that the SDK is out of date that isn’t driven by NuGet audit (and doesn’t depend on the project needing a PackageDownload).

 from the discussion in https://github.com/dotnet/sdk/issues/44421.

Reply:
The spec should explain this. To me, it currently reads like we're going to leave gaps in auditing. It should be clear that customers will get different notifications about these packages.